--- main/libmpdemux/frequencies.c.orig	2005-09-01 09:30:16.000000000 +0200
+++ main/libmpdemux/frequencies.c	2005-08-25 08:15:40.000000000 +0200
@@ -809,6 +809,54 @@
     FREQ_UHF,
 };
 
+static struct CHANLIST pal_switzerland[] = {
+    { "K32", 559250 }, // BBC Prime
+    { "S13", 246250 }, // CNN Int.
+    { "K34", 575250 }, // CNBC Europe/NBC Europe
+    { "K21", 471250 }, // Arte/KiKa
+    { "K05", 175500 }, // SF 1
+    { "K10", 211000 }, // SF 2
+    { "K35", 583250 }, // SFinfo
+    { "S23", 319250 }, // EuroNews
+    { "S17", 274500 }, // Eurosport (d + e)
+    { "S05", 133250 }, // ORF 1
+    { "S15", 260250 }, // ORF 2
+    { "S10", 168500 }, // 3sat
+    { "S19", 288500 }, // SAT 1
+    { "K09", 203750 }, // ARD
+    { "S04", 126250 }, // kabel eins
+    { "K29", 535250 }, // BR
+    { "K27", 519250 }, // PRO 7
+    { "K30", 543250 }, // VOX
+    { "K06", 182750 }, // RTL
+    { "K22", 479250 }, // RTL 2
+    { "S26", 343250 }, // Super RTL
+    { "S31", 383250 }, // WDR
+    { "K11", 218000 }, // ZDF
+    { "K24", 495250 }, // DSF
+    { "K31", 551250 }, // TV5 Europe
+    { "K28", 527250 }, // n-tv
+    { "S09", 161500 }, // Tele Züri
+    { "S14", 253250 }, // Südwest Fernsehen BW
+    { "S07", 147250 }, // Star TV
+    { "S33", 399250 }, // U1 TV STATION
+    { "S20", 295750 }, // MTV
+    { "S24", 327250 }, // VIVA
+    { "S11", 232000 }, // HSE24
+    { "S12", 239250 }, // TF 1
+    { "S28", 359250 }, // France 2
+    { "K07", 189750 }, // TSR 1
+    { "K37", 599250 }, // TSR 2
+    { "K25", 503250 }, // Canale 5
+    { "S08", 154500 }, // RAI UNO
+    { "K08", 196750 }, // TSI 1
+    { "S18", 281500 }, // TSI 2
+    { "K12", 225000 }, // TVE Int.
+    { "S16", 267500 }, // cablecom infokanal 
+    { "S27", 351250 }, // 3sat
+    { "S06", 140250 }, // Tele M1**
+};
+
 /* --------------------------------------------------------------------- */
 
 static struct CHANLIST pal_newzealand[] = {
@@ -1153,6 +1201,7 @@
     { "europe-west",      europe_west,       CHAN_COUNT(europe_west)       },
     { "europe-east",      europe_east,       CHAN_COUNT(europe_east)       },
     { "italy",	          pal_italy,         CHAN_COUNT(pal_italy)         },
+    { "switzerland",	  pal_switzerland,   CHAN_COUNT(pal_switzerland)   },
     { "newzealand",       pal_newzealand,    CHAN_COUNT(pal_newzealand)    },
     { "australia",        pal_australia,     CHAN_COUNT(pal_australia)     },
     { "ireland",          pal_ireland,       CHAN_COUNT(pal_ireland)       },
