Index: setup.py
===================================================================
--- setup.py	(revision 8238)
+++ setup.py	(working copy)
@@ -110,20 +110,12 @@
 # only add files not in share and src
 
 data_files = []
-# add some files to Docs
-for f in ('COPYING', 'ChangeLog', 'INSTALL', 'README'):
-    data_files.append(('share/doc/freevo-%s' % version.__version__, ['%s' % f ]))
-data_files.append(('share/doc/freevo-%s' % version.__version__, ['Docs/CREDITS' ]))
 data_files.append(('share/fxd', ['share/fxd/webradio.fxd']))
 
 # copy freevo_config.py to share/freevo. It's the best place to put it
 # for now, but the location should be changed
 data_files.append(('share/freevo', [ 'freevo_config.py' ]))
 
-# add docbook style howtos
-os.path.walk('./Docs/installation', docbook_finder, data_files)
-os.path.walk('./Docs/plugin_writing', docbook_finder, data_files)
-
 # start script
 scripts = ['freevo']
 
